src/instructlab/sdg/generate_data.py

```diff
@@ -183,12 +183,17 @@ def _context_init(
     num_instructions_to_generate: int,
     batch_num_workers: Optional[int],
     batch_size: Optional[int],
+    checkpoint_dir: Optional[str] = None,
+    save_freq: Optional[int] = 1,
 ):
     extra_kwargs = {}
     if batch_size is not None:
         extra_kwargs["batch_size"] = batch_size
         extra_kwargs["batch_num_workers"] = batch_num_workers
 
+    extra_kwargs["checkpoint_dir"] = checkpoint_dir
+    extra_kwargs["save_freq"] = save_freq
```

Comment:
You shouldn't need to use extra_kwargs for this ... it's only for args that are set conditionally based on batch_size?